### ISSUE TITLE
Clustername validation in Openshift installer on Openstack

### ISSUE THREAD

**Issue by sputmayer** [openshift/installer#6301]
4.8

```console
$ openshift-install version
openshift-install 4.8.5
built from commit 54c7628be380fcb568262dd49a4636da2e0baa21
release image <internal-registry>/installer/openshift-release-image@sha256:7047acb946649cc1f54d98a1c28dd7b487fe91479aa52c13c971ea014a66c8a8
```

# Platform:
openstack 

# Install Type
UPI 

# What happened?
Getting the following error when running the install
"cluster name is too long, please restrict it to 14 characters".

# What you expected to happen?
A way to bypass this issue or to understand why this validation exists on Openstack platform, so as to better explain the architecture team to fix the naming convention so that it sits within the max length.
Enter text here.

# How to reproduce it (as minimally and precisely as possible)?
Create a cluster on Openstack with metadata.name in install-config.yaml with more than 14 character.
For example
apiVersion: v1
baseDomain: openshift.example.com
compute:
- architecture: amd64
  hyperthreading: Enabled
  name: worker
  platform: {}
  replicas: 0
controlPlane:
  architecture: amd64
  hyperthreading: Enabled
  name: master
  platform: {}
  replicas: 3
metadata:
  creationTimestamp: null
  name: cluster-with-name-longer-than-14-chars

```console
$ your-commands-here
```

# Anything else we need to know?

Enter text here.

# References

- enter text here.

**Comment by mandre** [comment]
Hi @sputmayer, this should be fixed in master when https://github.com/openshift/installer/pull/6309 lands. We'll backport to 4.11 and 4.10 at a minimum.

**Comment by mandre** [comment]
This is fixed in 4.10 (#6420), 4.11 (#6355), and the upcoming 4.12 (#6309).
/close
